src/testdir/test_expr.vim

```diff
@@ -583,4 +585,14 @@ func Test_expr_eval_error()
   call assert_fails("let v = -{}", 'E728:')
 endfunc
 
+" Test for float value comparison
+func Test_float_compare()
+  call assert_true(1.2 == 1.2)
+  call assert_true(1.0 != 1.2)
+  call assert_true(1.2 > 1.0)
+  call assert_true(1.2 >= 1.2)
+  call assert_true(1.0 < 1.2)
+  call assert_true(1.2 <= 1.2)
```

Comment:
How about testing:
* NaN (which is never equal to anything, including itself)
* +infinity
* -infinity
* +0.0 and -0.0 (which should be equal despite having different representation)